docs/contributor/testing.md

```diff
@@ -0,0 +1,105 @@
+
+## Testing Levels
+
+| Test suite | Testing level | Purpose                                                                                                                                                                                       |
+|------------|---------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Unit       | Unit          | This test suite tests the units in isolation. It assesses the implementation correctness of the units of business logic.                                                                      |
+| Env-tests  | Integration   | This test suite tests the behaviour of the NATS Manager in integration with a Kubernetes API server replaced with a test double. It assesses the integration correctness of the NATS Manager. |
+| E2E        | Acceptance    | This test suite tests the usability scenarios of the NATS Manager in a cluster. It assesses the functional correctness of the NATS Manager.                                                   |
+
+> **NOTE:** The validation and defaulting rules are tested within the integration tests.
+
+### Unit tests and Env-tests
+
+To run the unit and integration tests, the following command needs to be executed. If necessary, the needed binaries for the integration tests are downloaded to `./bin`.
+Further information about integration tests can be found in the [Kubebuilder book](https://book.kubebuilder.io/reference/envtest.html).
+
+   ```sh
+   make test-only
+   ```
+
+### E2E tests
+
+Because E2E tests need a Kubernetes cluster to run on, they are separate from the remaining tests.
+
+1. Ensure you have the Kubecontext pointing to an existing Kubernetes cluster.
+
+2. Execute the E2E test.
+
+   If NATS Manager has not yet been deployed on the cluster, run:
+
+   ```sh
+   make e2e IMG=<container-registry>/nats-manager:<tag>
+   ```
+
+   Otherwise, simply run:
+
+   ```sh
+   make e2e-only
+   ```
+
+3. To adjust the log level, use the environment variable `E2E_LOG_LEVEL`.
+   The accepted values are `debug`, `info`, `warn`, and `error`.
+
+   To set the level, enter:
+
+   ```sh
+   export E2E_LOG_LEVEL="<loglevel>"
+   ```
+
+The E2E test consists of four consecutive steps. If desired, you can run them individually.
+
+1. Ensure you have the Kubecontext pointing on an existing cluster and NATS Manager is deployed.
+
+2. Set up the NATS CR and ensure all related resources (Pods, Services, PVCs) are set up.
+
+   ```sh
+   make e2e-setup
+   ```
+
+3. Execute a [bench test](https://docs.nats.io/using-nats/nats-tools/nats_cli/natsbench) on the NATS Server.
+
+   ```sh
+   make e2e-bench
+   ```
+
+   > **NOTE:** Running this step on slow hardware (like CI systems or k3d clusters) will result in poor performance.
+   > However, this tool is helpful to show that NATS JetStream is in an operational configuration.
```

Comment:
AFAIK, `make e2e-bench` only tests the core NATS.

Reply:
I think it will test jetstream, due to the `--js` flag being set [here](https://github.com/kyma-project/nats-manager/blob/2e098751a14cb6a6122f7d7368f3c2eb54cf7f8d/e2e/scripts/natsbench.sh#L25)